### PR TITLE
[11.0][mis_builder] Implement hide_always

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
Allows to hide kpis when this flag 'hide_always' is set in the assigned style.

### CODE REVIEW

File: mis_builder/tests/test_render.py [hunk 1]
```diff
@@ -209,6 +212,7 @@ def test_merge(self):
         self.assertEquals(style_props, {
             'color': '#00FF00',
             'dp': 1,
+            'hide_always': False,
         })
```
Comment (sbidoul) [inline mis_builder/tests/test_render.py]: why do you need to change this test?

**Comment by JordiBForgeFlow** [comment]
We're not going to use this feature. So closing the PR for now.

**Comment by JordiBForgeFlow** [comment]
I reconsidered and we'll use this feature :)

**Comment by sbidoul** [comment]
This PR should work, right?

**Comment by JordiBForgeFlow** [comment]
Yes, it works. I have been using it to prepare the financial ratios for
spain

**Comment by sbidoul** [comment]
Cool!

**Comment by JordiBForgeFlow** [comment]
@sbidoul See the financial ratios proposed for Spain. https://github.com/OCA/l10n-spain/pull/871 Perhaps you might be interested to use a similar approach for BE, LU

**Comment by sbidoul** [comment]
Nice! So you are overriding the reports to add new lines with the ratios, correct?

**Comment by JordiBForgeFlow** [comment]
I had to duplicate the reports from l10n_es_mis_report, not override.

Even when the xmlids are the same, since the reports belong to a different
module the complete xmlid is different.

I duplicated the balance sheet and profit and loss, hiding all the lines so
that the ratio kpis can refer to them.

Either you do that, or you directly add the ratios to the BS and P&L, but
that can be inconvenient if you want to distribute these reports, without
the ratios.

**Comment by sbidoul** [comment]
I see. I'll dig this idea of having the formulas of one report available in the computations of another.

**Comment by sbidoul** [comment]
Have you seen my comment about the changed test above? Except that this PR looks good to me.

**Comment by JordiBForgeFlow** [comment]
Yes, I attended your comment

**Comment by sbidoul** [comment]
I don't understand. The test is still modified.

**Comment by JordiBForgeFlow** [comment]
Hmm will check

File: mis_builder/tests/test_render.py (hunk 1 above)
Comment (JordiBForgeFlow) [reply mis_builder/tests/test_render.py]: 'hide_always' does not have a corresponding 'hide_always_inherit', and correspondingly will be returned here https://github.com/OCA/mis-builder/blob/11.0/mis_builder/models/mis_report_style.py#L157

Comment (JordiBForgeFlow) [reply mis_builder/tests/test_render.py]: Perhaps I should add 'hide_always_inherit'?

Comment (sbidoul) [reply mis_builder/tests/test_render.py]: I was wondering too

Comment (JordiBForgeFlow) [reply mis_builder/tests/test_render.py]: Hmm, now that I think about it, when you hide a higher level it should always hide lower levels. So I will create 'hide_always_inherit'.

**Comment by JordiBForgeFlow** [comment]
@sbidoul I have now added 'hide_always_inherit' and makes total sense.

**Review by sbidoul — APPROVED** [review]
Thanks! I'll merge this in 10.0 and port to 9 and 11

**Comment by sbidoul** [comment]
Merged at c3f41b933ddf4da6fac4362aecb4e2b4d33e9e0a

**Comment by sbidoul** [comment]
Ported to 11 and 9

**Pull request closed without merging**